### PR TITLE
Update Dynamic Client Registration API reference

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** "Fixed the JSON request in the "Register new client" POST example. Removed "refresh_token" from grant types because the current JSON request doesn't work." <br>
Re-created this PR from https://github.com/okta/okta-developer-docs/pull/2350 Thanks for original PR, @patrickjburke245 <br>
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* n/a

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/oauth-clients/index.md [hunk 1]
```diff
@@ -62,8 +62,7 @@ curl -v -X POST \
      "id_token"
   ],
   "grant_types": [
-     "authorization_code",
-     "refresh_token"
```
Comment (larsjohansen-okta) [inline packages/@okta/vuepress-site/docs/reference/api/oauth-clients/index.md:66]: We should not remove this, we should instead add `implicit` to the list of grant types.

**Review by larsjohansen-okta — CHANGES_REQUESTED** [review]

**Comment by huanliu-okta** [comment]
@brianduffield-okta have you tested this to be working?

**Comment by brianduffield-okta** [comment]
@huanliu-okta I did a simple test with the Postman collection (which includes `implicit` in the Create Client API call), and it works. Removing `implicit` also returned an error. @larsjohansen-okta may have done other testing. Is this enough to update docs?

**Review by huanliu-okta — APPROVED** [review]
Just want to make sure we tested the change. LGTM.

**Review by larsjohansen-okta — APPROVED** [review]

**Pull request merged**